accepted/PSR-4-autoloader-examples.md

```diff
@@ -117,11 +118,14 @@ class Psr4AutoloaderClass
     /**
      * Register loader with SPL autoloader stack.
      *
+     * @param bool $prepend If true, prepend the autoloader to the stack
+     * instead of appending it.
      * @return void
      */
-    public function register()
+    public function register($prepend = false)
```

Comment:
I guess we can type the argument to `bool` ?

Reply:
@drupol you mean `spl_autoload_register(array($this, 'loadClass'), false, (bool) $prepend);` ?

Reply:
No, this:

```suggestion
    public function register(bool $prepend = false)
```

Reply:
@drupol it's not PHP5-compatible code

Reply:
And it must be PHP5 compatible ? #shocked

Reply:
@drupol  lack of typing indicates it

Reply:
Do we have the official confirmation that PHP5 needs to be supported?

If yes, please ignore my comment.

Reply:
@drupol I'm don't see reasons to not support PHP5. Anyway, it's not for me to decide.

Reply:
Could be a long thread but basically, I don't see the reason to support something which is [no more supported officially](https://www.php.net/eol.php)... or else, what will be the limit?
I think adding types really help a lot of people when programming and avoid also many issues in advance.
But like you said, it's not up to us to decide, and it would be nice to have the point of view of someone who does.

Reply:
This is example code.  If we decide we can non-trivially touch it at all, we should go all in and update it to modern code.  Hell, I'd include union return types where appropriate, too, at that point.

That's if we decide we can touch it non-trivially at all to begin with; it's not obvious if this is part of the spec (which has vastly higher standards for what you can do) or not.

Reply:
@Crell @drupol without flame please

Reply:
I shut up and listen.